docs/source/development/design/defining_new_models.md

```diff
@@ -452,34 +475,78 @@ You should end up with something like this:
 def __init__(
     self,
     data: Data,
+    core_components: CoreComponents,
     update_interval: pint.Quantity,
     pond_data_path: Path,
     constants: FreshwaterConstants,
-    **kwargs: Any,
+    static: bool = False,
 ):
 
     # Call the __init__() method of the base class
-    super().__init__(data, update_interval, **kwargs)
+    super().__init__(data, core_components, static)
 
-    # Store model specific details as attributes.
-    self.pond_data_path = pond_data_path
+    # Type and document attributes
+    self.pond_data_path: Path
+    """Path to the pond data file."""
+    self.constants: FreshwaterConstants
+    """Constants for the model."""
 
-    # Store the constants relevant to the freshwater model
-    self.constants = constants
+    # Conditionally run setup steps.
+    if self._run_setup:
+        self._setup(pond_data_path=pond_data_path, constants=constants)
 
     # Save attribute names to be used by the __repr__
     self._repr.append("pond_data_path")
 ```
 
-#### Model dependencies
+### The `_setup` method
 
-Your model may depend on a particular execution order for other models. This order is
-found automatically by Virtual Ecosystem based on the variables that the models require
-to be initialised and updated. Eg. if a model requires variable `A` to be initialised
-and that variable is provided by another model, this second model will run first.
+The `_setup` method typically contains the bulk of the code that needs to run to setup
+the initial state of the model and populate the data variables listed in the
+`_vars_populated_by_init` attribute. The signature of the function typically takes the
+model specific arguments defined on the `__init__` method and uses those values to
+populate model attributes and calculate data values. It is typical for `_setup` to call
+additional methods that you define on the class or functions from additional submodules.
 
-If a suitable order cannot be found, the simulation will stop and an error message will
-be provided informing on the specific issue.
+Following the example above:
+
+```{code-block} ipython3
+def _setup(self, pond_data_path: Path, constants: FreshwaterConstants) -> None:
+    """Set up the freshwater model."""
+
+    self.pond_data_path = pond_data_path
+    self.constants = constants
+
+    # Load the data using an extra method on the class
+    self.load_pond_data(self.pond_data_path)
```

Comment:
I think this is an unfortunate example. You shouldn't want data to be loaded in a setup method...